### PR TITLE
Update footer styling to match spec

### LINKED ISSUE #1151: Footer does not match spec
Font size is currently 13px, spec says it should be 14px across all breakpoints. 

Other specs should also be checked.

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Update footer styling to match spec

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/footer/
- Cross ref with [spec](https://github.com/ubuntudesign/vanilla-design/blob/master/Footer) and verify example matches design

## Details

Fixes #1151

### CODE REVIEW

**Comment by barrymcgee** [comment]
Demo: http://vanilla-framework-1151-footer-spec.demo.haus/vanilla-framework/

**Comment by yaili** [comment]
Design review notes:
The bottom padding in L screen should be 24px, but right now it's 16px from the footer container plus 24px from the inline links. Can we remove the bottom padding on the links, and increase the footer container bottom padding to 24px (16px in S screen)?

File: scss/_patterns_footer.scss [hunk 1]
```diff
@@ -3,13 +3,15 @@
 // Footer pattern styling
 @mixin vf-p-footer {
   .p-footer {
-    border-top: 1px dotted $color-x-dark;
-    font-size: $sp-small;
+    border-top: 1px solid $color-mid-light;
+    color: $color-dark;
```
Comment (anthonydillon) [inline scss/_patterns_footer.scss]: We need to be careful when setting this color as we are not setting the background color. As this is currently the body color anyway I would suggest removing it.

File: scss/_patterns_footer.scss [hunk 2]
```diff
@@ -18,14 +20,18 @@
...
+      padding: $sp-small 0 $sp-medium;
 
       @media (min-width: $breakpoint-medium) {
         margin-top: 0;
-        padding: 0;
+        padding: $sp-medium 0 $sp-large;
       }
     }
 
+    nav {
```
Comment (anthonydillon) [inline scss/_patterns_footer.scss]: Should target only classname in patterns.

File: scss/_patterns_footer.scss [hunk 3]
```diff
@@ -42,7 +48,6 @@
     &__link {
       border-bottom: 0;
       color: $color-dark;
```
Comment (anthonydillon) [inline scss/_patterns_footer.scss]: Same here with the color

File: scss/_patterns_footer.scss (hunk 1 above)
Comment (barrymcgee) [inline scss/_patterns_footer.scss]: I only added it as the spec is explicit - if we remove it here, we should remove it there @yaili?

File: scss/_patterns_footer.scss (hunk 2 above)
Comment (barrymcgee) [inline scss/_patterns_footer.scss]: Note, this will make the next release a minor.

File: scss/_patterns_footer.scss (hunk 3 above)
Comment (barrymcgee) [inline scss/_patterns_footer.scss]: By not setting a color here, they will be the default link color, which is blue.

**Comment by yaili** [comment]
The issue is still there:
- [ ] p-footer__links should have no bottom padding
- [ ] p-footer should have 24px bottom padding on L screen, 16px on S screen

**Comment by barrymcgee** [comment]
@yaili Demo didn't update, it should now be updated.

File: scss/_patterns_footer.scss (hunk 3 above)
Comment (anthonydillon) [inline scss/_patterns_footer.scss]: I see, @yaili could we set the background color of the footer pattern?

Comment (yaili) [inline scss/_patterns_footer.scss]: @anthonydillon I think the footer should work on a white or grey background without setting colour

**Pull request merged**